### PR TITLE
Health check

### LINKED ISSUE #204: [BE Feat] Create health check route
## Checklist
Before submitting this ticket, make sure this ticket:
- [ ] Is **labelled properly** - if this is a refactor, add the label!
- [ ] Has the **SPUR project assigned to it**.
- [ ] Has an **assigned milestone**.

And lastly:
- [x] Once someone picks up this ticket, **make sure to assign it**.

## Description
server status endpoint, reports system info

### PR DESCRIPTION
**Pull request by juancwu**
## Checklist
Before opening this PR, make sure the PR:
- [x] Has an **assignee or group of assignees**.
- [x] Is **labelled properly**.
- [x] Has the **SPUR project assigned to it**.
- [x] Has an **assigned milestone**.

Additionally, make sure that:
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

## Description
Add a health check route. There are some other files changes to allow the app the run since everything is broken right now.

## Linked Issues
- Got PR before issue so :shrug: 

## Type of Change
- [x] New feature (non-breaking change which adds functionality).
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Testing
- `internal/v1/v1_health/healthcheck_test.go` tests internal methods that gather health information
- `internal/tests/server_test.go` blackbox testing of the health check route

### CODE REVIEW

File: backend/internal/server/types.go [hunk 1]
```diff
@@ -1 +1,36 @@
...
+
+import (
+	"KonferCA/SPUR/storage"
+	"time"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/labstack/echo/v4"
+)
+
+type Server struct {
```
Comment (aidantrabs) [inline backend/internal/server/types.go:10]: isnt this defined in the the `v1_heatlh` package already?

File: backend/internal/v1/v1_auth/auth.go [hunk 2]
```diff
@@ -0,0 +1 @@
+package v1auth
```
Comment (aidantrabs) [inline backend/internal/v1/v1_auth/auth.go:1]: how come you're not using the actual package name? `package v1_auth` works

see here - https://go.dev/ref/spec#Package_clause

File: backend/internal/v1/v1_health/healthcheck.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	"context"
+	"net/http"
+	"runtime"
+	"time"
+
+	"github.com/labstack/echo/v4"
+)
+
+/*
+V1 healthchekc endpoint handler.
```
Comment (aidantrabs) [inline backend/internal/v1/v1_health/healthcheck.go]: typo lol

Comment (juancwu) [reply backend/internal/v1/v1_health/healthcheck.go]: shhhhh

File: backend/internal/v1/v1_auth/auth.go (hunk 2 above)
Comment (juancwu) [reply backend/internal/v1/v1_auth/auth.go:1]: I just pressed enter to whatever the editor suggested lol

File: backend/internal/server/types.go (hunk 1 above)
Comment (juancwu) [reply backend/internal/server/types.go:10]: It is?

Comment (juancwu) [reply backend/internal/server/types.go:10]: Different, there is a handler but not the Server struct, which should be in the server package.

Comment (aidantrabs) [reply backend/internal/server/types.go:10]: oh no i didnt mean the server struct, i meant the types relating to the DB

i.e `DatabaseInfo`, `SystemInfo` and `HealthReport`, they're already defined in `v1_health` and I believe should not be needed here

**Review by AmirAgassi — APPROVED** [review]

**Pull request merged**